servant-auth-server/src/Servant/Auth/Server/Internal/JWT.hs

```diff
@@ -47,37 +47,33 @@ class ToJWT a where
 jwtAuthCheck :: FromJWT usr => JWTSettings -> AuthCheck usr
 jwtAuthCheck config = do
   req <- ask
-  token <- maybe mempty return $ do
-    authHdr <- lookup "Authorization" $ requestHeaders req
-    let bearer = "Bearer "
-        (mbearer, rest) = BS.splitAt (BS.length bearer) authHdr
-    guard (mbearer `constTimeEq` bearer)
-    return rest
+  authHdr <- maybe (fail "Missing auth header") pure $
+    lookup "Authorization" $ requestHeaders req
+  either (fail . T.unpack) (parseJWT config) $ parseHeader authHdr
+
+parseJWT :: FromJWT usr => JWTSettings -> Token usr -> AuthCheck usr
```

Comment:
Could you rename this to something that makes clear that's it's doing verification (e.g. `verifyAndDecodeJWT`)?